utils/githooks/README.md

```diff
@@ -1,13 +1,86 @@
-# Git hooks
+# About DAOS Git hooks
 
-This directory contains a githooks framework that can be employed locally to automate things like
-copyright updates on modified files.
+Githooks are a [well documented](https://git-scm.com/docs/githooks) feature
+of git that enable various local exectubles to be run during various stages of
+the git workflow.
 
-It supports adding local hooks by placing your custom hook in
-`utils/githooks/<hook>.d/<num>-user-<name>` files.
+The DAOS repo contains several githooks built-in githooks that are intended
+to help developers conform to DAOS community coding standards and practices
+and to avoid some common mistakes in the development cycle.
 
-To use the commit hooks here, do the following locally, or alternatively copy the files into place.
+To enable these standard githooks requires a two step process:
+
+1. Install the hooks
+
+Configure your core.hookspath as follows (Recommended):
 
 ```sh
-git config core.hookspath utils/githooks
+git config.hookspath utils/githooks
 ```
+
+Additionally, one can copy the files into an already configured path.
+
+With the first option, any new githooks added to the repository will
+automatically run, but possibly require additional software to produce the
+desired effect.  Additionally, as the branch changes, the githooks
+change with it.
+
+2. Install all of the required tools
+
+The Githooks framework in DAOS is such that the hooks will all run.
+However, some hooks will simply check for required software and are
+effectively a noop if such is not installed.
+
+On many systems, the required packages can be installed through standard means
+but customization may be required.  The following packages are used by
+built-in githooks.
+
+1. clang-format version 14.0.5 or higher.  If the check is unable to parse
+the version output, it will fail.  Try running
+`<root>/site_scons/site_tools/extra/extra.py` to check.
+2. pylint
+3. flake8
+4. yamllint
```

Comment:
These all come from python, `pip -r requirements.txt` I think is the recommended way of ensuring they're installed.

Reply:
An optional requirement is the `gh` command